magpie/api/webhooks.py

```diff
@@ -63,16 +92,20 @@ def send_webhook_request(webhook_config, params, update_user_status_on_error=Fal
     :param update_user_status_on_error: update the user status or not in case of a webhook error
     """
     # Replace each instance of template parameters if a corresponding value was defined in input
-    for template_param in WEBHOOK_TEMPLATE_PARAMS:
-        if template_param in params:
-            for k, v in webhook_config["payload"].items():
-                webhook_config["payload"][k] = v.replace("{" + template_param + "}", params[template_param])
-
     try:
-        resp = requests.request(webhook_config["method"], webhook_config["url"], data=webhook_config["payload"])
+        for template_param in WEBHOOK_TEMPLATE_PARAMS:
+            if template_param in params:
+                webhook_config["payload"] = replace_template(template_param,
```

Comment:
I'm wondering if you're not replacing the original config here? Does the config still contain the template placeholder after this function call?

Reply:
I think this would replace the input parameter value and also the variable in the preceding function that called this one. It doesn't affect the value saved in the registry though.
I guess it is safer to make a copy at the beginning of this function, so the parameter won't be affected!

Reply:
The final refactoring is perfect!